pgmax/fg/graph.py

```diff
@@ -396,25 +308,25 @@ class FactorGraphState:
     """FactorGraphState.
 
     Args:
-        variable_group: A variable group containing all the variables in the FactorGraph.
+        variable_groups: VariableGroups in the FactorGraph.
         vars_to_starts: Maps variables to their starting indices in the flat evidence array.
             flat_evidence[vars_to_starts[variable]: vars_to_starts[variable] + variable.num_var_states]
             contains evidence to the variable.
         num_var_states: Total number of variable states.
         total_factor_num_states: Size of the flat ftov messages array.
-        named_factor_groups: Maps the names of named factor groups to the corresponding factor groups.
+        factor_groups: FactorGroups in the FactorGraph
         factor_type_to_msgs_range: Maps factors types to their start and end indices in the flat ftov messages.
         factor_type_to_potentials_range: Maps factor types to their start and end indices in the flat log potentials.
         factor_group_to_potentials_starts: Maps factor groups to their starting indices in the flat log potentials.
         log_potentials: Flat log potentials array concatenated for each factor type.
         wiring: Wiring derived for each factor type.
     """
 
-    variable_group: groups.VariableGroup
-    vars_to_starts: Mapping[nodes.Variable, int]
+    variable_groups: Sequence[groups.VariableGroup]
+    vars_to_starts: Mapping[Tuple[Any, int], int]
     num_var_states: int
     total_factor_num_states: int
-    named_factor_groups: Mapping[Hashable, groups.FactorGroup]
+    factor_groups: Tuple[groups.FactorGroup, ...]
```

Comment:
Looks like this `factor_groups` is only used to check whether a given `factor_group` exists. If so it's redundant (you can just check whether a given `factor_group` is `in factor_group_to_potential_starts`)